pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1798,7 +1810,17 @@ func handleDelete(ctx *domainContext, key string, status *types.DomainStatus) {
 	publishDomainStatus(ctx, status)
 
 	if status.Activated {
+		if status.HasError() {
+			log.Functionf("handleDelete(%v) clearing existing error for %s",
+				status.UUIDandVersion, status.DisplayName)
+			status.ClearError()
+			publishDomainStatus(ctx, status)
+		}
 		doInactivate(ctx, status, true)
+		if status.HasError() {
+			log.Errorf("doInactivate failed: Aborting the delete of %s", status.DisplayName)
+			return
```

Comment:
Is this helpful? The DomainConfig is gone but you keep the domain and DomainStatus around forever?
Thus if someone recreates the same DomainConfig odd things will happen.

Reply:
With these changes, we are making sure that the domain is gone. If the domain is not deleted from the device then we are avoiding the deletion of ```DomainStatus```

>Thus if someone recreates the same DomainConfig odd things will happen.

That's true. this thing is already happening with the current code. After deleting the instance, the task is not going away and the next instance with the same name is running with some bogus configuration. After this fix, we will be able to identify such a case.